superset/datasets/commands/create.py

```diff
@@ -49,10 +50,14 @@ def run(self) -> Model:
             # Updates columns and metrics from the dataset
             dataset.fetch_metadata(commit=False)
             db.session.commit()
-        except (SQLAlchemyError, DAOCreateFailedError) as ex:
+        except (SQLAlchemyError, DAOCreateFailedError, SupersetSecurityException) as ex:
             logger.warning(ex, exc_info=True)
             db.session.rollback()
-            raise DatasetCreateFailedError() from ex
+            raise DatasetCreateFailedError(
+                *{"message": str(ex)}
+                if isinstance(ex, SupersetSecurityException)
```

Comment:
@john-bodley Overwrite the message only if the error is a SupersetSecurityException, since other cases are just "Create failed" and the DatasetCreateFailedError message is more informative.

```
  - {'message': 'Dataset could not be created.'}
  + {'message': 'Create failed'}
```

Reply:
@justinpark I think you'll get the same behavior just by catching the SupersetSecurityException and let DatasetCreateFailedError raise as it is now. I don't believe you'll need to change the error message there in line 55. Unless I'm missing something else?

Reply:
@eschutho I hope to overwrite the error message when SupersetSecurityException is thrown because it specifies the error description.
`Dataset could not be created.` (DatasetCreateFailedError) vs `Only "SELECT" statements are allowed` (SupersetSecurityException)